roles/init_cp/tasks/main.yml

```diff
@@ -13,10 +13,12 @@
   command: >-
     kubeadm init --config {{ kubeadm_config_yaml }}
     {% if enable_kubeadm_patches -%}
-      {% if target_kube_version is version("1.22", ">=") -%}
-      --patches
-      {%- else -%}
-      --experimental-patches
+      {% if target_kube_version is version("1.23", "<") -%}
+        {% if target_kube_version is version("1.22", ">=") -%}
```

Comment:
version("1.23", "<") and version("1.22", ">=") are quite "opposite" test, If in-yaml works in 1.22 (https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/#kubeadm-k8s-io-v1beta3-InitConfiguration, documentation is rather unclear) , I would jump on it directly removing one condition here.
If only supported in 1.23, I would do 
```
{% target_kube_version is version("1.23", ">=") %}
{% elif  target_kube_version is version("1.22", "==") %} {# or '>=' if '==' doesn't work #}
   -- patches
{% elif  target_kube_version is version("1.19", ">=") %}
   --experimental-patches
{% endif %}
```
Which could also fix 1.18 version that does'nt support patches (which is a bug ^^)